src/store.cc

```diff
@@ -671,11 +654,7 @@ StoreEntry::forcePublicKey(const cache_key *newkey)
 {
     if (StoreEntry *e2 = (StoreEntry *)hash_lookup(store_table, newkey)) {
         assert(e2 != this);
-        debugs(20, 3, "Making old " << *e2 << " private.");
-
-        // TODO: check whether there is any sense in keeping old entry
-        // shareable here. Leaving it non-shareable for now.
-        e2->setPrivateKey(false);
+        debugs(20, 3, "releasing clashing " << *e2);
         e2->release(false);
```

Comment:
I would even make e2->release(true) here: if there is another entry with the same key, there could be attached clients to it, and we should not prevent them from normal finishing, making the entry unshareable.

Reply:
I share your sentiment but I am a little worried that we have chosen to add a TODO instead of using a true parameter in 39fe14b. Your call, but my recommendation is to leave the TODO in place, to be addressed in a future dedicated commit. If you decide to address and remove this TODO now, then please do not forget to reflect the changes in the PR description.

Reply:
I left this old TODO. Apart from this fix, we may need to revise and fix other places with release(true) as well.